src/workers/postUpdated.ts

```diff
@@ -255,10 +255,25 @@ const updatePost = async ({
   content_type = PostType.Article,
 }: UpdatePostProps) => {
   const postType = contentTypeFromPostType[content_type];
-  const databasePost = await entityManager
+  let databasePost = await entityManager
     .getRepository(postType)
     .findOneBy({ id });
 
+  // If we don't find the post, we need to check if it's a youtube video and
+  // try to find it again as an article
+  if (!databasePost && content_type === PostType.VideoYouTube) {
+    await entityManager.getRepository(ArticlePost).update(
+      { id },
+      {
+        type: PostType.VideoYouTube,
+      },
+    );
+
+    databasePost = await entityManager
+      .getRepository(postType)
+      .findOneBy({ id });
+  }
+
```

Comment:
@capJavert @rebelchris I changed it to this, as it failed trying to set `videoId` on `ArticlePost`, I have to update the post type, then re-fetch it.

Could also make this a bit more generic. So before we get the databasePost in the first place, we just do a query like @capJavert suggested, and update the type (it it is allowed for that type)

Reply:
This is where I wonder if we can't get the type in the initial one if we change to `Post` and then use post.type === the one we expect?
Might be a little bit more stable?

Reply:
We can but we can't write it since the columns would be just the ones existing in `Post`. Raw query is more memory efficient since no need to assemble entity but it would do the same thing. No strong preferences.

Reply:
```ts
await entityManager
  .createQueryBuilder()
  .update(Post)
  .set({ type: PostType.VideoYouTube })
  .where('id = :id', { id })
  .execute();
```

Reply:
Like this ☝️ @idoshamun?

Reply:
Updated to use query builder for updating